scripts/functions/filter-intovalue-criteria.R

```diff
@@ -0,0 +1,77 @@
+# We often limit `trials` to trials meeting criteria per intovalue.
+# These functions apply the appropriate filters.
+
+
+# trials ------------------------------------------------------------------
+
+read_iv_trials <- function(){
+
+  # Read in trials
+  trials_filepath <- here::here("data", "processed", "trials.csv")
+  cli::cli_alert_info("Read in trials from {.file {trials_filepath}}")
+  trials_raw <- readr::read_csv(trials_filepath, show_col_types = FALSE)
+
+  cli::cli_alert_info("Trials filtered for intovalue criteria: completion date, status, interventional, German UMC lead, prefer IV2>1")
+
+  # Apply the IntoValue exclusion criteria
+  # https://github.com/quest-bih/clinical-dashboard/blob/main/prep/prep-intovalue.R#L9-L18
+  trials <-
+
+    trials_raw |>
+    dplyr::filter(
+      iv_completion,
+      iv_status,
+      iv_interventional,
+      has_german_umc_lead,
+      ## In case of dupes, exclude IV1 version
+      !(is_dupe & iv_version == 1)
+    ) |>
+    # Check that trns are unique (duplicates removed)
+    assertr::assert(assertr::is_uniq, id)
+
+}
+
+
+# cross-registrations -----------------------------------------------------
+
+read_iv_cross_registrations <- function(){
+
+  # Read in cross-registrations
+  cross_registrations_filepath <- here::here("data", "processed", "trn", "cross-registrations.rds")
+  cli::cli_alert_info("Read in cross-registrations from {.file {cross_registrations_filepath}}")
+  cross_registrations_raw <- readr::read_rds(cross_registrations_filepath)
+
+  # Read in trials (needed to filter)
+  trials <- suppressMessages(read_iv_trials())
+
+  cli::cli_alert_info("Cross-registrations filtered for intovalue")
+
+  cross_registrations <-
+    cross_registrations_raw |>
+
+    # Remove exact duplicates (not sure why these appear)
+    dplyr::distinct() |>
+
+    # Limit to cross-registrations of trials meeting IntoValue exclusion criteria
+    dplyr::semi_join(trials, by = "id") |>
+
+    # Limit to cross-registrations found using IV2 paper
+    dplyr::semi_join(trials, by = c("id", "doi"))
+
+  stopifnot(
+
+    # Check that no trn/doi's in trials and removed from crossreg
+    trials |>
+      dplyr::anti_join(cross_registrations, by = c("id", "doi")) |>
+      dplyr::semi_join(cross_registrations_raw, y = _, by = c("id", "doi")) |>
```

Comment:
@lucfernandez: following up here on the error you emailed about. @delwen looping you in since you replied to the email.

My first hunch is that this has something to do with the base R pipe and underscore placeholder, which you can read about in this post: https://tidyverse.org/blog/2023/04/base-vs-magrittr-pipe/ My first step would be to check/update R, RStudio, and packages. Because my mac is old and on OS11 per Apple recommendation, I'm not running the newest RStudio. Your issue prompted me to update my R and packages though. Now I am running (and you can use this code to check what you're running:

``` r
getRversion()
#> [1] '4.4.1'
rstudioapi::versionInfo()$version
#> [1] ‘2023.12.1.402’
```

Reply:
I just tried again with my all freshened up stack and it's running:

```r
# Get trials and cross-registrations, limited to those meeting intovalue criteria
source(here::here("scripts", "functions", "filter-intovalue-criteria.R"))
trials <- read_iv_trials()
#> ℹ Read in trials from /Users/maia/Documents/Repositories/intovalue-data-lucfernandez/data/processed/trials.csv
#> ℹ Trials filtered for intovalue criteria: completion date, status, interventional, German UMC lead, prefer IV2>1
cross_registrations <- read_iv_cross_registrations()
#> ℹ Read in cross-registrations from /Users/maia/Documents/Repositories/intovalue-data-lucfernandez/data/processed/trn/cross-registrations.rds
#> ℹ Cross-registrations filtered for intovalue   
```

Reply:
Thank you, I updated R and everything is looking good!